src/primitives/bezier.jl

```diff
@@ -0,0 +1,103 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    BezierCurve(points)
+
+A recursive Bézier curve with control points `points`.
+See https://en.wikipedia.org/wiki/Bézier_curve.
+A point on the curve `b` can be evaluated by calling
+`b(t)` with `t` between `0` and `1`.
+The evaluation method defaults to DeCasteljau's algorithm
+for accurate evaluation. Horner's method, faster with a
+large number of points but less precise, can be used via
+`b(t, Horner())`.
+
+## Example
+
+```julia
+BezierCurve(Point2[(0.,0.),(1.,-1.)])
+```
+"""
+struct BezierCurve{Dim,T,V<:AbstractVector{Point{Dim,T}}} <: Primitive{Dim,T}
+  controls::V
+end
+
+BezierCurve(points::Vararg) = BezierCurve(collect(points))
+BezierCurve(points::AbstractVector{<:Tuple}) = BezierCurve(Point.(points))
+
+ncontrols(b::BezierCurve) = length(b.controls)
+
+degree(b::BezierCurve) = ncontrols(b) - 1
```

Comment:
Just double checking, degree is the most accurate term right? It is not order or something else...

Reply:
Hmm I actually find both terms to be used. Although I quickly googled "order vs degree", and found that the order of a polynomial is the number of terms in a series expansion, and so it equals the degree + 1. So, here it is the degree, and `ncontrols` actually equals the order.

Reply:
Yes, just wanted to double check. I think degree is perfectly fine 💯